File: 백준/Silver/3273. 두 수의 합/두 수의 합.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
    cin.tie(0)->sync_with_stdio(0);

    int n;
    cin>>n;

    int arr[n];
    vector<int>a(n);
    for (int i=0; i<n; i++) {
        cin>>arr[i];
    }
    int x;
    cin>>x;

    sort(arr,arr+n);
    int left=0,right=n-1,cnt=0;
    while (left<right) {
        int sum=arr[left]+arr[right];
        if (sum==x) {
            cnt++;
            right--;
        }else if (sum>x) {
            right--;
        }else {
            left++;
        }
    }
    cout<<cnt;
}

File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
import sys
input=sys.stdin.readline

n=int(input())
arr=list(map(int,input().split()))
x=int(input())

arr.sort()
l=0
r=n-1
cnt=0

while l<r:
    sum=arr[l]+arr[r]
    if sum==x:
        cnt+=1
        r-=1
    elif sum<x:
        l+=1
    else:
        r-=1
print(cnt)